matchbox_socket/src/error.rs

```diff
@@ -1,9 +1,24 @@
 use crate::webrtc_socket::error::SignalingError;
 
-/// Errors that can happen when using Matchbox.
+/// Errors that can happen when using Matchbox sockets.
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
-    /// An error occurring during the signaling loop.
-    #[error("An error in the signaling loop: {0}")]
-    Signaling(#[from] SignalingError),
+    /// An error occurring if the connection fails to establish. Perhaps check your connection or try again.
+    #[error("The connection failed to establish. Check your connection and try again.")]
+    ConnectionFailed {
+        /// The source of the connection failure
+        source: SignalingError,
+    },
+    /// Unexpected fatal error ocurred with messaging. Please file an issue or triage.
+    #[error("An unexpected error ocurred at runtime with messaging: {source}")]
+    Runtime {
+        /// The source of the connection failure
+        source: SignalingError,
+    },
```

Comment:
I don't think this error should exist. I feel like it's better to just panic/error at the point where our assumptions were wrong, that would allow us or users to simply run with `RUST_BACKTRACE=1` and get a proper stack trace, fix the bug.

But perhaps I'm in the minority. How do you feel about this @garryod ?

Reply:
This is caused only by UnknownCommunication, e.g. a peer starts sending binary when we only accept text through the socket.

I could rename it, or we could downgrade the error to a `warn!` and remove the error. (my preference)

Reply:
Sounds good!

Reply:
removed